ddp-workspace/projects/ddp-dsm-ui/src/app/dashboard-statistics/dashboard-statistics.component.ts

```diff
@@ -1,66 +1,134 @@
-  ) {
+  /**
+   * Add here name of the statistics you want to be selected
+   * and initialized for the first time
+   */
+  private activeTab: StatisticsName = 'charts';
+  private isDateChanged = false;
+
+  private readonly statisticsSubject$: Subject<void> = new Subject<void>();
+  private readonly statisticsSubjectSubscription$: Subscription;
+
+  constructor(private dashboardStatisticsService: DashboardStatisticsService, private errorService: ErrorsService) {
+    this.statisticsSubjectSubscription$ = this.statisticsSubject$
+      .pipe(
+        tap(() => this.loading = true),
+        map(() => this.enumeratedActiveTab),
+        switchMap((statisticsEnum: StatisticsEnum) =>
+          this.dashboardStatisticsService.getStatisticsFor(statisticsEnum, this.dateRange)
+            .pipe(
+              take(1),
+              catchError(this.handleError.bind(this)),
+              finalize(() => this.loading = false)
+            )
+        )
+      )
+      .subscribe({next: this.initializeData.bind(this)});
   }
 
   ngOnInit(): void {
-    this.initData();
+    this.statisticsSubject$.next();
+  }
+
+  ngOnDestroy(): void {
+    this.statisticsSubjectSubscription$.unsubscribe();
   }
 
-  private initData(): void {
-    this.hasRequiredRole = this.roleService.allowedToViewEELData();
-    this.Charts = this.dashboardStatisticsService.ChartFactory(this.dateRange)
-      .pipe(catchError(this.catchErrorAndReturnArray.bind(this)), finalize(() => this.loading = false));
-    this.Counts = this.dashboardStatisticsService.Counts;
+  public getStatisticsFor({tab}: MatTabChangeEvent): void {
+    this.activeTab = tab.ariaLabel as StatisticsName;
+    if(this.allowStatisticsUpdate) {
+      this.statisticsSubject$.next();
+      this.isDateChanged = false;
+    }
   }
 
-  public dateChanged(dateRange: DateRangeModel): void {
-    console.log(dateRange, 'DATE - dashboard.statistics');
+  public dateChanged(dateRange: IDateRange): void {
     this.loading = true;
+    this.isDateChanged = true;
     this.dateRange = dateRange;
-    this.initData();
+    this.statisticsSubject$.next();
   }
 
-  get getConfiguration(): any {
-    return {
-      responsive: true,
-      displaylogo: false
-    };
+  public retry(): void {
+    this.loading = true;
+    this.statisticsSubject$.next();
   }
 
-  private catchErrorAndReturnArray(error: HttpErrorResponse): Observable<never> {
-    if(error instanceof HttpErrorResponse) {
-      this.errorMessage.next(error);
-    }
+  public get selectedTabIndex(): number {
+    return this.statisticsCollection.findIndex(statistics => statistics.name === this.activeTab);
+  }
+
+  private get allowStatisticsUpdate(): boolean {
+    return (!this.activeTabStatObj.data || this.isDateChanged || this.errorHas[this.activeTab]) && !this.loading;
+  }
+
+  private get enumeratedActiveTab(): StatisticsEnum {
+    return StatisticsEnum[this.activeTab.slice(0, this.activeTab.length - 1).toUpperCase()];
```

Comment:
**_#optional#tip_**

`a.slice(0, -1)` is a short version of `a.slice(0, a.length-1)`

Reply:
Sure! Fixed